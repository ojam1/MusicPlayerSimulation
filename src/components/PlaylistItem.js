import React, { Component } from 'react';
import { connect } from 'react-redux';

import {
  removeSongFromPlaylist,
  currentPlayingSong
} from '../actions/playlist';

class PlaylistItem extends Component {
  render() {
    const { song, currentPlayingSong, removeSongFromPlaylist } = this.props;
    const { artist, album, title } = song;

    return (
      <div>
        <div>Artist: {artist}</div>
        <div>Title: {title}</div>
        <div>Album: {album}</div>
        <button onClick={() => currentPlayingSong(song)}>Play</button>
        <button onClick={() => removeSongFromPlaylist(song)}>
          Remove song from playlist
        </button>
      </div>
    );
  }
}

export default connect(
  null,
  { removeSongFromPlaylist, currentPlayingSong }
)(PlaylistItem);
