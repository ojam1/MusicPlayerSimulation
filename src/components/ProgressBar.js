import React, { Component } from 'react';
import { connect } from 'react-redux';

export class ProgressBar extends Component {
  componentDidMount() {
    this.setState(() => {
      interval: setInterval(() => this.tick(), 1000)
    });
  }

  constructor(props) {
    super(props);
    this.state = {
      currentPostion: 0,
      interval: 0
    };
  }

  convertToSeconds() {
    const { duration } = this.props.playingSong.playingSong;

    return 10;
  }

  tick() {
    if (this.props.playingSong.isPlaying) {
      this.setState(state => ({
        currentPostion: state.currentPostion + 1
      }));

      if (this.state.currentPostion === this.convertToSeconds()) {
        clearInterval(this.state.interval);
      }
      console.log('this.convertToSeconds: ', this.convertToSeconds());
      console.log('this.state.currentPostion: ', this.state.currentPostion);
      console.log('this.props.playingSong: ', this.props.playingSong);
    }
  }
  render() {
    return (
      <div className='progress'>
        <div
          className='progress-bar progress-bar-striped progress-bar-animated'
          role='progressbar'
          style={{ width: this.state.currentPostion }}
          aria-valuemax={this.convertToSeconds().toString()}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  playingSong: state.currentSong
});

export default connect(
  mapStateToProps,
  null
)(ProgressBar);
