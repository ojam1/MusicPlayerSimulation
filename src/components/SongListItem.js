import React, { Component } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { addSongToPlaylist } from '../actions/playlist';


const Button = styled.button`
background: purple;
border: none;
color: white;
font-weight: 500;
padding: 1.2rem;
`;

const SongText = styled.div`
  text-align: left;
  font-weight: bold;
`;

class SongListItem extends Component {
  render() {
    const { song, addSongToPlaylist } = this.props
    const { artist, album, title } = song;

    return (
      <div>
        <SongText>Artist: {artist}</SongText>
        <SongText>Title: {title}</SongText>
        <SongText>Album: {album}</SongText>
        <Button onClick={() => addSongToPlaylist(song)}>
          Add to playlist
        </Button>
      </div>
    );
  }
}

export default connect(null, { addSongToPlaylist })(SongListItem);
