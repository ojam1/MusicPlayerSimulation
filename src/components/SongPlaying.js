import React, { Component } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { changePlayingState, currentPlayingSong } from '../actions/playlist';

const Button = styled.button`
background: purple;
border: none;
color: white;
font-weight: 100;
padding: 1rem;
`;

const SongText = styled.div`
  text-align: left;
`;

const Header = styled.h1`
  font-weight: bold;
`;
const Divider = styled.div`
width:5px;
height:auto;
display:inline-block;
`;


class Playing extends Component {
  handlePrevious() {
    const { playlist, currentPlayingSong } = this.props;
    const { playingSong } = this.props.playingSong;

    if ((playlist.indexOf(playingSong)) - 1 >= 0) {
      currentPlayingSong(playlist[playlist.indexOf(playingSong) - 1]);
    }
  }

  handleNext() {
    const { playlist, currentPlayingSong } = this.props;
    const { playingSong } = this.props.playingSong;

    if ((playlist.indexOf(playingSong)) + 1 !== playlist.length) {
      currentPlayingSong(playlist[playlist.indexOf(playingSong) + 1]);
    }
  }

  renderPlayPauseButton() {
    const { changePlayingState } = this.props;
    const { isPlaying } = this.props.playingSong;

    if (isPlaying) {
      return (
        <Button onClick={() => changePlayingState(!isPlaying)}>
          Pause
          </Button>
      );
    }
    return (
      <Button onClick={() => changePlayingState(!isPlaying)}>
        Play
        </Button>
    );
  }

  render() {
    const { artist, album, title } = this.props.playingSong.playingSong;

    return (
      <div>
        <Header>Now Playing</Header>
        <SongText>Artist: {artist}</SongText>
        <SongText>Title: {title}</SongText>
        <SongText>Album: {album}</SongText>
        <Button onClick={() => this.handlePrevious()}>Previous</Button>
        <Divider />
        {this.renderPlayPauseButton()}
        <Divider />
        <Button onClick={() => this.handleNext()}>Next</Button>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  playingSong: state.currentSong,
  playlist: state.playlist.songsInPlaylist
});

export default connect(mapStateToProps, { changePlayingState, currentPlayingSong })(Playing);
