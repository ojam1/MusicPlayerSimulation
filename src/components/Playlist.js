import React, { Component } from 'react';
import { connect } from 'react-redux';
import uuid from 'uuid/v4';

import PlaylistItem from './PlaylistItem';

export class Playlist extends Component {
  render() {
    const { playlist } = this.props;

    return (
      <div>
        <h1>Playlist</h1>
        <div>
          {playlist.length == 0 ? (
            <h2>No songs in playlist</h2>
          ) : (
            playlist.map(song => <PlaylistItem key={uuid()} song={song} />)
          )}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  playlist: state.playlist.songsInPlaylist
});

export default connect(
  mapStateToProps,
  null
)(Playlist);
