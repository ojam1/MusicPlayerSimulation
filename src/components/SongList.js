import React, { Component } from 'react';
import { connect } from 'react-redux';
import SongListItem from './SongListItem';

import { allSongs } from '../actions/songs';

export class SongList extends Component {
  componentWillMount() {
    this.props.allSongs();
  }

  render() {
    return (
      <div>
        <h1>Available Songs</h1>
        <div>
          {this.props.songs.map(song => (
            <SongListItem key={song.id} song={song} />
          ))}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  songs: state.song
});

export default connect(
  mapStateToProps,
  { allSongs }
)(SongList);
