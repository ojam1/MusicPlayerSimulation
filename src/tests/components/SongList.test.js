import { shallow } from 'enzyme';
import React from 'react';

import { SongList } from '../../components/SongList';

test('renders correctly', () => {
  const mockAllSongs = jest.fn();

  let songs = [
    {
      id: 1,
      artist: 'Erika McTerry',
      title: 'id nulla ultrices',
      album: 'tincidunt in',
      duration: 3.25
    },
    {
      id: 2,
      artist: 'Aaren Allender',
      title: 'amet consectetuer adipiscing elit',
      album: 'ultrices posuere cubilia',
      duration: 1.85
    }
  ];

  const wrapper = shallow(<SongList songs={songs} allSongs={mockAllSongs}/>);

  expect(wrapper).toMatchSnapshot();
});