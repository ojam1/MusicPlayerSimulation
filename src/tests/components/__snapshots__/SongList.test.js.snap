// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Available Songs
  </h1>
  <div>
    <Connect(SongListItem)
      key="1"
      song={
        Object {
          "album": "tincidunt in",
          "artist": "Erika McTerry",
          "duration": 3.25,
          "id": 1,
          "title": "id nulla ultrices",
        }
      }
    />
    <Connect(SongListItem)
      key="2"
      song={
        Object {
          "album": "ultrices posuere cubilia",
          "artist": "Aaren Allender",
          "duration": 1.85,
          "id": 2,
          "title": "amet consectetuer adipiscing elit",
        }
      }
    />
  </div>
</div>
`;
